tests/templates/test_subroutines/test_qubitization.py

```diff
@@ -0,0 +1,335 @@
+        diff_method = "backprop" if shots is None else "parameter-shift"
+        qnode = qml.QNode(self.circuit, dev, interface="torch", diff_method=diff_method)
+
+        params = torch.tensor(self.params, requires_grad=True)
+        jac = torch.autograd.functional.jacobian(qnode, params)
+        assert qml.math.shape(jac) == (4,)
+        assert qml.math.allclose(jac, self.exp_grad, atol=0.01)
+
+    @pytest.mark.tf
+    @pytest.mark.parametrize("shots", [None, 50000])
+    @pytest.mark.xfail(reason="tf gradient doesn't seem to be working, returns ()")
+    def test_qnode_tf(self, shots):
+        """ "Test that the QNode executes and is differentiable with TensorFlow. The shots
+        argument controls whether autodiff or parameter-shift gradients are used."""
+        import tensorflow as tf
+
+        dev = qml.device("default.qubit", shots=shots, seed=10)
+        diff_method = "backprop" if shots is None else "parameter-shift"
+        qnode = qml.QNode(self.circuit, dev, interface="tf", diff_method=diff_method)
+
+        params = tf.Variable(self.params)
+        with tf.GradientTape() as tape:
+            res = qnode(params)
+
+        jac = tape.gradient(res, params)
+        assert qml.math.shape(jac) == (4,)
+        assert qml.math.allclose(res, self.exp_grad, atol=0.001)
+
+    @pytest.mark.xfail(reason="see https://github.com/PennyLaneAI/pennylane/issues/5507")
+    @pytest.mark.usefixtures("use_legacy_and_new_opmath")
+    def test_legacy_new_opmath_diff(self):
+        coeffs, ops = np.array([0.1, -0.3, -0.3]), [qml.X(0), qml.Z(1), qml.Y(0) @ qml.Z(2)]
+
+        dev = qml.device("default.qubit")
+
+        @qml.qnode(dev)
+        def circuit_dot(coeffs):
+            H = qml.dot(coeffs, ops)
+            qml.Qubitization(H, control=[3, 4])
+            return qml.expval(qml.PauliZ(0))
+
+        @qml.qnode(dev)
+        def circuit_Hamiltonian(coeffs):
+            H = qml.Hamiltonian(coeffs, ops)
+            qml.Qubitization(H, control=[3, 4])
+            return qml.expval(qml.PauliZ(0))
+
+        assert np.allclose(qml.grad(circuit_dot)(coeffs), qml.grad(circuit_Hamiltonian)(coeffs))
+
+
+def test_copy():
+    """Test that a Qubitization operator can be copied."""
+
+    H = qml.dot([1.0, 2.0], [qml.PauliX(0), qml.PauliZ(1)])
+
+    orig_op = qml.Qubitization(H, control=[2, 3])
+    copy_op = copy.copy(orig_op)
+    assert qml.equal(orig_op, copy_op)
+
+    # Ensure the (nested) operations are copied instead of aliased.
+    assert orig_op is not copy_op
+    assert orig_op.hyperparameters["hamiltonian"] is not copy_op.hyperparameters["hamiltonian"]
+    assert orig_op.hyperparameters["control"] is not copy_op.hyperparameters["control"]
+
+
+def test_map_wires():
+    """Test that a Qubitization operator can be mapped to a different wire mapping."""
+
+    H = qml.dot([1.0, 2.0], [qml.PauliX(0), qml.PauliZ(1)])
+
+    op = qml.Qubitization(H, control=[2, 3])
+    op2 = op.map_wires({0: 5, 1: 6, 2: 7, 3: 8})
```

Comment:
What is the expected behaviour for Hamiltonian? do we map its wires as a side effect of this?

Reply:
The Hamiltonian is not changed (I based my decision on `TrotterProduct`)